feat: Add `appstore review-list`

kenstir opened the pull request:
Fixes #609 

* Add `as review-list --id=appid` to list reviews.  For now, it uses a simple
  text output that is readable in the terminal.

blacktop commented:
are there going to be other sub-cmds other than `ls` ?

kenstir commented:
> are there going to be other sub-cmds other than `ls` ?

No I don't think so.  I just started using that cmd "in anger" yesterday, and I'm using it to email me a list of all reviews in the last 24h.
```
for appid in $app1 $app2 $app3; do
    ipsw appstore review ls --since '25h' --id $appid
done
```

But it's not helpful to list apps by app ID so I think I want another command `app ls` to list apps, so I could do
```
for appid in `ipsw appstore app ls --show=appid`; do
    ...
```

I see there is already an unused function `GetApps()` so I would start there.

blacktop commented:
I have no apps w/ reviews. Does this work for you?

kenstir commented:
Sure does!  I use it in a cron job to mail me any reviews in the last 25 hours like so:
```
kenstir@trinity ipsw (feat/as-review)
$ ./ipsw --config-quiet appstore review ls --since '25h' --id $appid

kenstir@trinity ipsw (feat/as-review)
$ ./ipsw --config-quiet appstore review ls --after='2025-01-01' --id $appid

-------------------
Jan 21 2025 [★★★★★] by emjewell21
Lists. Android has them.
    (responded Dec 21 2024)

-------------------
Jan  7 2025 [★★★★ ] by MelLifeLongLearner
Due Date Alert
    (responded Jan  4 2025)

2 reviews since Jan  1 2025 00:00:00
2 responses

To respond, visit https://appstoreconnect.apple.com/apps/1400443994/distribution/activity/ios/ratingsResponses
```

blacktop commented on cmd/ipsw/cmd/appstore/appstore_review_ls.go:
```diff
@@ -0,0 +1,158 @@
+/*
+Copyright © 2024 Kenneth H. Cox
```
this should be the `ipsw` Copyright

kenstir replied:
Done

blacktop commented on pkg/appstore/review.go:
```diff
@@ -0,0 +1,118 @@
+	queryParams := url.Values{}
+	queryParams.Add("include", "response")
+	queryParams.Add("sort", "-createdDate")
+	//queryParams.Add("exists[publishedResponse]", "false")
```
should this be removed?

kenstir replied:
Done

blacktop commented on cmd/ipsw/cmd/appstore/appstore_review_ls.go:
```diff
@@ -0,0 +1,158 @@
+
+// ASReviewListCmd represents the appstore review ls command
+var ASReviewListCmd = &cobra.Command{
+	Use:           "ls",
```
if there is only 1 sub-cmd, there there shouldn't be a sub-cmd

kenstir replied:
Good point, done

kenstir commented:
All review comments have been addressed @blacktop and it's ready for another review.  Importantly, I removed the unnecessary sub-cmd `ipsw as review ls` and renamed it `ipsw as review-list`, modeled after `device-list`.

blacktop commented:
LGTM
